피드 레이아웃 및 프래그먼트의 간단한 구성

bsgreentea opened the pull request:
피드 리스트를 볼 수 있는 프래그먼트와 레이아웃을 구성했습니다.

<img width="219" alt="KakaoTalk_20201106_214549740" src="https://user-images.githubusercontent.com/46317084/98371020-d5cecc80-207e-11eb-9fd3-6a89a993782f.png">


코드가 산으로 가는 것 같아 일단 간단하게 테스트해보는 것으로 끝냈습니다ㅜㅜ

bsgreentea committed:
Add layouts #8 레이아웃 구성

- my page와 feed item 구성

bsgreentea committed:
기타 구성 #8

- 사진 둥글게 하기

bsgreentea committed:
Merge branch 'develop' into feature/#8_compose_feed

bsgreentea committed:
Add resources #8

리소스 추가 및 레이아웃 조정

bsgreentea committed:
Merge branch 'develop' into feature/#8_compose_feed

bsgreentea committed:
Implement FeedsFragment #8

- 피드 레이아웃 수정
- 피드 출력 확인을 위한 간단한 코드 구성

bsgreentea committed:
Modify NavigationController #8

- 내비게이션 바 수정
- color 리소스 추가

bsgreentea commented on app/src/main/res/values/colors.xml:
```diff
@@ -9,4 +9,5 @@
     <color name="white">#FFFFFFFF</color>
```
rgb에 8자리 hex를 사용하면 앞의 두 자리가 투명도를 나타내기 때문에 투명도가 필요 없는 상황이라면 6자리로 나타내는게 오버드로우를 줄일 수 있다고 하네요! 혹시 의도한 부분인지 궁금합니다

지난번에 미처 언급을 못했었네요..

ChoMk replied:
아아 white_opa80와의 차이를 두려고 그랬던건데 오버드로우가 있는지는 처음 알았네요.. 아무래도 앞자리 보고서 투명도를 확인하니..

ChoMk replied:
이 친구도 수정해서 같이 밀어 넣어주세요 ㅎㅎ

ChoMk commented on app/build.gradle:
```diff
@@ -35,6 +35,10 @@ android {
     }
+
+    buildFeatures {
+        dataBinding = true
```
혹시 
dataBinding {
        enabled = true
    } 
이거랑 겹치는거 아닌가요??

bsgreentea replied:
이전에 한번 당기면서 체크를 못했네요 ㅎㅎ.. 수정해서 올리겠습니다!

ChoMk commented on app/src/main/java/com/example/bikini_android/ui/feeds/FeedsFragment.kt:
```diff
@@ -0,0 +1,56 @@
+ * @author bsgreentea
+ */
+
+class FeedsFragment : BaseFragment() {
```
해당 부분은 mainholder activity위에 올라갈 fragment라서 mainholder fragment가 적절해 보이네요 ㅎㅎ..

bsgreentea replied:
앗 넵ㅎㅎ

ChoMk commented on app/src/main/java/com/example/bikini_android/ui/holder/BottomNavigationItem.kt:
```diff
@@ -23,9 +23,14 @@ enum class BottomNavigationItem(
-        menuId = R.id.adding_feed_icon,
-        navigationAction = { navigateController -> navigateController.navigateToAddingFeed() }
+
+    //        ADDING_CONTENTS(
```
테스트 용도였어서 그냥 adding contents 관련헤서 지워서 정리 한번 밀어 넣어 주세용
다음에 추가 컨태츠 로직 작업 때 다시 만들어도 되니깐

ChoMk commented on app/src/main/res/values/dimens.xml:
```diff
@@ -5,9 +5,10 @@
   -->
-
 <resources>
-    <!--common-->
```
혹 주석을 지운 이유가 있나요? 해당 부분은 dimens 파일 안에서 구조적으로 값들을 구분할 방법이 없어서 주석을 통하여 구분하려고했던거라..

ChoMk replied:
위처럼 만들면 공통이 아니어도 dimens에 들어가서 사용성이 떨어질 수는 있지만(공통된 친구들만 dimens에 들어간게 아니라..) 하지만 의도한 스타일 이릉이 명료하니깐 xlm을 읽을 때 가독성이 좀 더 좋아질거 같네요..(네이밍을 잘해야하지만..)

bsgreentea replied:
당기면서 의식하지 못한채로 지워버린 것 같습니다.. 복구해놓겠습니다 ㅎㅎ 언급해주시니까 차이가 확실히 느껴지네요

ChoMk commented on app/src/main/res/layout/item_feed.xml:
```diff
@@ -0,0 +1,64 @@
+    <androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:paddingBottom="20dp">
```
혹시 20 패딩이 왜 들어 간지 알 수 있을가요?

bsgreentea replied:
간격을 두는게 좋아보여서 그랬습니다. 특별한 이유는 없었는데 잘 사용되지 않는 방법인가요?!

ChoMk replied:
음 공통으로 사용할 리사이클러뷰에 들어갈 config 용 친구를 하나 만들어야 할거 같네요
리사이클러뷰 아이템 사이에 들어갈 마진이나 혹은 레이아웃 메니저가 들어갈 수 있는..
지금 당장은 아니지만 하나의 아이템에 다음 아이템과의 간격을 고려하여 마진이나 패딩이 들어가면은 그 아이템을 다른 곳에서 새로운 패딩과 마진을 넣어주기 힘들어질 수 있어서용

qwebnm7788 replied:
아니면 ItemDecoration 을 사용해도 좋을 것 같네요

ChoMk replied:
ㅇㅇ 데코레이션이나 콘피그 파일을 따로 하나 만드는건 이슈 등록해서 따로 진행하면 될거 같네요.

ChoMk commented on app/src/main/res/layout/item_feed.xml:
```diff
@@ -0,0 +1,64 @@
+
+        <ImageView
+            android:id="@+id/user_thumbnail"
+            android:layout_width="50dp"
```
dp, sp는 dimens에 정리하여 넣어주면 좋을거 같아요

bsgreentea committed:
코드리뷰 반영 #8

- 중복되는 사항 제거
- 테스트용 코드 제거
- 색상 투명도 수정
- BaseFragment -> MainHolderFragment로 수정
- 높낮이 dimens 리소스 추가

bsgreentea committed:
Merge branch 'develop' into feature/#8_compose_feed

bsgreentea committed:
Update NavigationController.kt

bsgreentea commented on app/src/main/res/layout/fragment_user_page.xml:
```diff
@@ -0,0 +1,68 @@
+            android:layout_height="wrap_content"
+            android:layout_margin="20dp"
+            android:text="sample_id"
+            android:textSize="20sp"
```
이 부분은 정말 대략적인 틀만 만들었기 때문에 dimens resource로 빼놓기가 애매하더라고요..
추후 작업시에 정리, 수정하도록 하겠습니다
